File: frontend/src/types/driver-schema.ts
/**
 * Driver schema types for data source configuration
 */

export type FieldType = 'string' | 'integer' | 'boolean' | 'password' | 'file_path'

export interface FieldMetadata {
  name: string
  type: FieldType
  required: boolean
  default?: any
  description: string
  min_value?: number | null
  max_value?: number | null
  placeholder: string
}

export interface DriverSchema {
  driver: string
  display_name: string
  description: string
  category: 'database' | 'file'
  fields: FieldMetadata[]
}

export type DriverSchemas = Record<string, DriverSchema>

File: backend/app/models/driver_schema.py
"""API models for driver schema metadata."""

from typing import Any, Literal

from pydantic import BaseModel, Field


class FieldMetadataResponse(BaseModel):
    """Response model for field metadata.

    Describes a single configuration field for a data source driver.
    """

    name: str = Field(..., description="Field identifier")
    type: Literal["string", "integer", "boolean", "password", "file_path"] = Field(..., description="Field data type")
    required: bool = Field(..., description="Whether the field is mandatory")
    default: Any = Field(None, description="Default value for the field")
    description: str = Field("", description="Human-readable description")
    min_value: int | None = Field(None, description="Minimum value for integer fields")
    max_value: int | None = Field(None, description="Maximum value for integer fields")
    placeholder: str = Field("", description="Example value to show in UI")


class DriverSchemaResponse(BaseModel):
    """Response model for driver schema.

    Describes the complete configuration schema for a data source driver.
    """

    driver: str = Field(..., description="Driver identifier")
    display_name: str = Field(..., description="Human-readable driver name")
    description: str = Field(..., description="Driver description")
    category: Literal["database", "file"] = Field(..., description="Driver category")
    fields: list[FieldMetadataResponse] = Field(..., description="List of supported configuration fields")
